### DISCUSSION TITLE
OsmAnd versus Google Maps

### DISCUSSION THREAD

**Discussion by ignacio-agullo** [osmandapp/OsmAnd#15671] [General]
Hello there!  I am an enthusiast user of navigators that has just found the OsmAnd forum.  I have been using navigators in general for over 14 years and OsmAnd in particular for over 7 years.  I think navigators are a great invention and am grateful to the people who develops them in general and to OsmAnd developers in particular.  I got plenty of experiences to share, but as long as this is the OsmAnd forum and not my personal blog, I will focus on the one topic that I think is worth bringing up to attention: an analysis of OsmAnd versus Google Maps.  I suppose that you have already discussed Google Maps at some point, but I also see that Google Maps is not mentioned amongst the most recent 20 discussions on this forum, and yes, I noticed the link "I have done a search for similar discussions" at the bottom at the form and followed it to find no Google Maps mentions, so this might be the right time to bring it up again.

I want to ride my bycycle, I want to ride my bike... well I want to drive my car too, and to walk also, so I have been using both OsmAnd and Google Maps for all three purposes during the last seven years.  This is what I have found about them:

-Autonomy: Google Maps consumes significantly more battery than OsmAnd.  Driving in my car with my old Samsung Galaxy SIII hooked to the recharge cable, if I used OsmAnd the battery would still increase its charge, but if I used Google Maps the battery would reduce its charge until eventually the smartphone would shut down.  This autonomy problem would be aggravated by long times driving while recharging, or hot temperatures, both of which would make the smartphone so much hot as to the recharge being automatically stopped.  Either way, with the recharge enabled or disabled, the superior autonomy provided by OsmAnd would be crucial.  At present I use a Samsung Galaxy S10+ with faster recharge that allows the battery charge to increase with either OsmAnd and Google Maps working, so this advantage for OsmAnd is less important today.

-Offline Navigation: When I started using OsmAnd in 2015, Google Maps still didn't had Offline Navigation - it would gain it shortly after (though at the cost of introducing a Google account in your smartphone and let the Big Brother watch you).  So as to drive without data consumption, OsmAnd was the superior choice - and for driving in foreing countries with no roaming, OsmAnd was the only choice.  Now roaming is so much improved, it is always present all over the European Union, though there are still some zones with no coverage where offline navigation is still crucial.

-Bycycle Navigation: Both OsmAnd and Google Maps provide an altitude profile of the routes - in the case of OsmAnd, I guess that that is actually provided by OpenStreetMap.  That is another fantastic feature I am grateful for, it is fantastic for us bycycle lovers to know in advance where the big climbs are, and to be able to consider alternatives.  But them, both OsmAnd and Google Maps _ignore_ this information.  Let's say I am travelling from Vigo to Tui, and I need to pass across some hills.  Shall I take the route where I have to climb 300 meters?  Of course not, both OsmAnd and Google Maps will take me on steeper routes where I save 7% of the distance at the expense of climbing 33% more altitude.  This happened to me this very year.  No guys no, that route has a shorter distance but takes 30 more minutes.  Also, both OsmAnd and Google Maps takes me through cycloturism routes, even though they are not paved and I need to go slowly to avoid the stones in the way.  Well, I just happen not to make cycloturism, I just want to arrive to my destination through the faster route, be it my workplace or whatever.  At present, both OsmAnd and Google Maps are only good to show me the altitude profile of the routes before chosing them - after choosing one, I need to switch to Automobile Navigation so I am taken through paved roads rather than unpaved ones.

-Shortest route: I tested OsmAnd shortest route in the past, only to discover that it would take me across towns (and their traffic lights that slowed my down so much) instead of around them, and even take me along the slower side lane in the highway because the highway would take a right curve and the side lane would do a shorter distance when it joined back the highway.  Now there is no longer a 'shortest route' but a 'low consumption (short)' route, but frankly, I am afraid to try it.  As for Google Maps, it doesn't currently have this option.  I am scarred enough as to consider this choice masochism and not try it again soon.

-Usability: Google Maps always haves the advantage when it comes to destination.  During many years I couldn't use OsmAnd to travel to the village of Villaviudas (Palencia) - in order to find the place, I would need to use Google Maps.  Fortunately, now Villaviudas is present - this gap is also closing, and this time in favour of OsmAnd.  But Google Maps will probably continue to have more shops, malls and service stations on its database, as Google profits from advertising and all those are potential customers.  Up to 2016, OsmAnd had a pretty rigid interface that would require the user to introduce the country, region and location of the destination.  Around 2017, that changed to become more flexible - but for a while, I wan't able to find my destinations.  I would introduce the name of a city, only to infuriatingly find dozens of streets named after the city in the villages in the way.  This was a setback in usability.  After a while (and an update, I think), the interface changed again and since then I can (with an additional tap) switch to a mode where only locations names are searched for, so street names no longer obfuscate my search.

**Comment by ignacio-agullo** [comment]
Just a correction to make: The Android Google Maps app stills consumes power faster than the smartphone can recharge from the car.  It is just that I learned to use the trick of temporarily switching down the GPS - say, if Google Maps tells me to follow a road for 10 kilometers straight, I switch down the GPS and keep an eye on the kilometers count, then when the 10 kilometers are about to finish I switch on the GPS again, so the smartphone can recharge a bit.  This trick has kept me from trouble many times when the battery was low.

**Comment by ferromint** [comment]
> [...] Bycycle Navigation [...] provide an altitude profile of the routes [...] OsmAnd [...] ignore this information
> [...] travelling from Vigo to Tui [...] OsmAnd [...] take me on steeper routes where I save 7% of the distance at the expense of climbing 33% more altitude
> [...] I just want to arrive to my destination through the faster route [...]

Do you mean that even though you selected the `Use elevation` option, the corresponding route is only a little less, but still too hilly for your biking preference/stamina?

A feature request for this case would be to parametrize the impact of the elevation information in the offline routing engine in a more granular way than just on/off.

Calculating a bike route in OsmAnd 4.3.5 between the two locations results in the following two options:
<table>
<tr>
<td>use elevation: off</td>
<td>use elevation: on</td>
</tr>
<tr>
<td>distance: 29.8km</td>
<td>distance: 33.6km (+13%)</td>
</tr>
<tr>
<td>elevation gain: 876m</td>
<td>elevation gain: 458m (-48%)</td>
</tr>
<tr>
<td><img src="https://user-images.githubusercontent.com/122570460/218310510-3bd0926f-0417-4693-a508-86f1af90d419.jpg"/></td>
<td><img src="https://user-images.githubusercontent.com/122570460/218310482-4efa62a7-f060-4941-8472-44671bd5c148.jpg"/></td>
</tr>
</table>
